front/src/applications/operationalStudies/types.ts

```diff
@@ -74,3 +75,29 @@ export type TrainScheduleImportConfig = {
   startTime: string;
   endTime: string;
 };
+
+// Extraction of some required and non nullable properties from osrdEditoastApi's PathProperties type
+export type ManageTrainSchedulePathProperties = {
+  electrifications: {
+    /** List of `n` boundaries of the ranges.
+        A boundary is a distance from the beginning of the path in mm. */
+    boundaries: number[];
+    /** List of `n+1` values associated to the ranges */
+    values: (
+      | {
+          type: 'electrification';
+          voltage: string;
+        }
+      | {
+          lower_pantograph: boolean;
+          type: 'neutral_section';
+        }
+      | {
+          type: 'non_electrified';
+        }
+    )[];
+  };
+  geometry: GeoJsonLineString;
```

Comment:
You can also remove the comment above, the code is clearer now with this fix:
```suggestion
  electrifications: NonNullable<PathProperties['electrifications']>;
  geometry: NonNullable<PathProperties['geometry']>;
```

Reply:
Done. I don't know why but I tried first to use the NonNullable for the whole type with Partial<PathProperties> ... and didn't think of your suggestion which is great !